### ISSUE TITLE
Unhelpful traceback produced if peering vnet name is specified instead of resource id

### ISSUE THREAD

**Issue by stewartadam** [Azure/azure-cli#7016]
**Describe the bug**
When creating a vnet peer and specifying a vnet name instead of full resource id, an unhelpful traceback is produced. Help output is also not sufficiently clear that the input is a full path/resource ID instead of just the vnet name.

**To Reproduce**
Command:
```
az network vnet peering create  --name hub-to-spoke1 \
  --resource-group spoke2 --vnet-name spoke2-vnet \
  --remote-vnet-id hub-vnet \
  --allow-vnet-access --allow-forwarded-traffic --allow-gateway-transit
```

**Actual behavior**
```
'subscription'
Traceback (most recent call last):
  File "/usr/local/Cellar/azure-cli/2.0.43/libexec/lib/python3.7/site-packages/knack/cli.py", line 197, in invoke
    cmd_result = self.invocation.execute(args)
  File "/usr/local/Cellar/azure-cli/2.0.43/libexec/lib/python3.7/site-packages/azure/cli/core/commands/__init__.py", line 369, in execute
    six.reraise(*sys.exc_info())
  File "/usr/local/Cellar/azure-cli/2.0.43/libexec/lib/python3.7/site-packages/six.py", line 693, in reraise
    raise value
  File "/usr/local/Cellar/azure-cli/2.0.43/libexec/lib/python3.7/site-packages/azure/cli/core/commands/__init__.py", line 343, in execute
    result = cmd(params)
  File "/usr/local/Cellar/azure-cli/2.0.43/libexec/lib/python3.7/site-packages/azure/cli/core/commands/__init__.py", line 182, in __call__
    return self.handler(*args, **kwargs)
  File "/usr/local/Cellar/azure-cli/2.0.43/libexec/lib/python3.7/site-packages/azure/cli/core/__init__.py", line 436, in default_command_handler
    result = op(**command_args)
  File "/usr/local/Cellar/azure-cli/2.0.43/libexec/lib/python3.7/site-packages/azure/cli/command_modules/network/custom.py", line 2938, in create_vnet_peering
    aux_subscription = parse_resource_id(remote_virtual_network)['subscription']
KeyError: 'subscription'
```

**Expected behavior**
Simple error noting that an invalid resource id (e.g. `/subscriptions/...`) was not provided, and that the name alone is not sufficient.

Specifically: when `parse_resource_id` cannot parse the resource it should raise an error, and this error should be caught by the command handler to produce a better actionable error message.

**Environment summary**
Install Method (e.g. pip, interactive script, apt-get, Docker, MSI, edge build) / CLI version (`az --version`) / OS version / Shell Type (e.g. bash, cmd.exe, Bash on Windows)
macOS 10.13.6, installed via brew, bash shell,

```
azure-cli (2.0.43)

acr (2.1.2)
acs (2.2.2)
advisor (0.6.0)
ams (0.2.1)
appservice (0.2.1)
backup (1.2.0)
batch (3.3.1)
batchai (0.4.0)
billing (0.2.0)
cdn (0.1.0)
cloud (2.1.0)
cognitiveservices (0.2.0)
command-modules-nspkg (2.0.2)
configure (2.0.18)
consumption (0.4.0)
container (0.3.2)
core (2.0.43)
cosmosdb (0.2.0)
dla (0.2.0)
dls (0.1.0)
dms (0.1.0)
eventgrid (0.2.0)
eventhubs (0.2.1)
extension (0.2.1)
feedback (2.1.4)
find (0.2.12)
interactive (0.3.27)
iot (0.2.0)
keyvault (2.2.1)
lab (0.1.0)
maps (0.3.1)
monitor (0.2.1)
network (2.2.2)
nspkg (3.0.3)
policyinsights (0.1.0)
profile (2.1.1)
rdbms (0.3.0)
redis (0.3.0)
reservations (0.3.1)
resource (2.1.1)
role (2.1.2)
search (0.1.0)
servicebus (0.2.1)
servicefabric (0.1.0)
sql (2.1.1)
storage (2.1.1)
vm (2.2.0)

Python location '/usr/local/opt/python/bin/python3.7'
Extensions directory '/Users/profile/.azure/cliextensions'

Python (Darwin) 3.7.0 (default, Jun 29 2018, 20:13:13) 
[Clang 9.1.0 (clang-902.0.39.2)]
```

**Comment by stewartadam** [comment]
> I don't see how that could be any clearer without calling out what is disallowed

Given the majority of other CLI commands use RG/Name pair to specify a resource and every other parameter for *this* command uses name, my issue was just that it was not consistent with other params and very easy to miss. Simply including `Resource ID of the remote VNet (e.g. '/subscriptions/...') would make this abundantly clear.

**Comment by adewaleo** [comment]
Fixed in #7391.
